apps/projects/tasks.py

```diff
@@ -1,141 +1,219 @@
 import logging
-import os
 
-from bson import json_util
+from bson import json_util, ObjectId
 from flask import current_app as app
 from pymongo import ReturnDocument
 
 from celery.exceptions import MaxRetriesExceededError
-from lib.celery_app import celery
-from lib.utils import format_id, get_url_for_media
 from lib.video_editor import get_video_editor
 
+from celery_app import celery
+
 logger = logging.getLogger(__name__)
 
 
 @celery.task(bind=True, default_retry_delay=10)
-def task_edit_video(self, sdoc, updates, action='post'):
+def edit_video(self, project_json, changes):
     """
     Task use tool for edit video and record the data and update status after finished,
-    :param file_path: full path edit video
-    :param sdoc: type json string, data info edit video
-    :param updates: type dictionary, changes apply to the video
-    :param action: put/replace action for new edited video
-    :param retry:
-    :return:
+    :param project_json: project json
+    :param changes: changes apply to the video
     """
-    doc = json_util.loads(sdoc)
-    storage_id = doc['storage_id']
-    try:
-        video_stream = app.fs.get(doc['storage_id'])
 
+    project = json_util.loads(project_json)
```

Comment:
fyi it's possible to setup custom serialization so you wouldn't have to json encode/decode it always